lib/matplotlib/backends/backend_cairo.py

```diff
@@ -134,6 +134,13 @@ def _fill_and_stroke (self, ctx, fill_c, alpha, alpha_overrides):
 
     @staticmethod
     def convert_path(ctx, path, transform):
+        extents = ctx.clip_extents()
+        bbox = Bbox([[extents[0], extents[1]], [extents[2], extents[3]]])
+        try:
+            path = path.clip_to_bbox (bbox)
```

Comment:
Except when it doesn't and breaks things.

We have pep8 tests as part of the test suite, but some file (like this one) are black-listed as known-bad.